app/src/main/shared/components/PageWithHeader/PageWithHeader.tsx

```diff
@@ -77,7 +82,9 @@ export abstract class PageWithHeader<TLocationProps>
 }
 
 export interface PageProperties<T> {
-    location: Location<T>;
-    router: IRouter;
+    location?: Location;
+    router?: Router;
     onLoad?: (props:Partial<T>) => void;
+    match: match<T>;
+    history?: History;
```

Comment:
Are some of these properties optional to allow for the unit tests? If so, could you add a comment explaining that here. I assume it's not possible to mock the router in the tests?